### ISSUE TITLE
Checkpoint SNAPPY compression, incompatible for inference?

### ISSUE THREAD

**Issue by petteriTeikari** [tensorflow/tensorflow#5628]
As referenced in [https://github.com/tensorflow/tensorflow/issues/1669](https://github.com/tensorflow/tensorflow/issues/1669), I got the same error when trying to restore from self-finetuned checkpoint

> "Data loss: Unable to open table file /tmp/model.ckpt-4971: Data loss: corrupted compressed block contents: perhaps your file is in a different file format and you need to use a different restore operator? It's likely that your checkpoint file has been compressed with SNAPPY."

I fine-tuned the checkpoint `inception_resnet_v2_2016_08_30.ckpt` for my own images TF-Slim (modified [finetune_inception_v3_on_flowers.sh](https://github.com/tensorflow/models/blob/master/slim/scripts/finetune_inception_v3_on_flowers.sh)). The fine-tuning seems to go well, and I get good performance for the test error. But when I try to do inference using the newly created checkpoint I got the above-mentioned same error, the same error happening with with `inspect_checkpoint` -tool.

The demo inference from there works fine as well: [https://github.com/tensorflow/models/blob/master/slim/slim_walkthough.ipynb](https://github.com/tensorflow/models/blob/master/slim/slim_walkthough.ipynb)

In other words, how to save the checkpoints with compatible compression? And how is it possible to save checkpoints on my computer (Ubuntu 14.04, TF 0.11.0rc2, CUDA 8.0) which are not possible to open a moment later?

**Comment by petteriTeikari** [comment]
I added that line to to slim.learning.train in [train_image_classifier.py](https://github.com/tensorflow/models/blob/master/slim/train_image_classifier.py)

```
saver=tf.train.Saver(write_version=1)
```

And got the `inspect_checkpoint` working, however now I got the following error with `freeze_graph`:

```Python
Traceback (most recent call last):
  File "/home/petteri/tensorflow/bazel-bin/tensorflow/python/tools/freeze_graph.runfiles/org_tensorflow/tensorflow/python/tools/freeze_graph.py", line 135, in <module>
    tf.app.run()
  File "/home/petteri/tensorflow/bazel-bin/tensorflow/python/tools/freeze_graph.runfiles/org_tensorflow/tensorflow/python/platform/app.py", line 43, in run
    sys.exit(main(sys.argv[:1] + flags_passthrough))
  File "/home/petteri/tensorflow/bazel-bin/tensorflow/python/tools/freeze_graph.runfiles/org_tensorflow/tensorflow/python/tools/freeze_graph.py", line 132, in main
    FLAGS.output_graph, FLAGS.clear_devices, FLAGS.initializer_nodes)
  File "/home/petteri/tensorflow/bazel-bin/tensorflow/python/tools/freeze_graph.runfiles/org_tensorflow/tensorflow/python/tools/freeze_graph.py", line 104, in freeze_graph
    _ = tf.import_graph_def(input_graph_def, name="")
  File "/home/petteri/tensorflow/bazel-bin/tensorflow/python/tools/freeze_graph.runfiles/org_tensorflow/tensorflow/python/framework/importer.py", line 258, in import_graph_def
    op_def = op_dict[node.op]
KeyError: u'DenseToDenseSetOperation'
```

**Comment by girving** [comment]
@alexalemi Can you take a look?

**Comment by EncodeTS** [comment]
I met the same problem.

**Comment by petewarden** [comment]
That freeze_graph error seems to indicate that you're using an older version of TensorFlow to load the graph, where 'DenseToDenseSetOperation' was still in contrib?

We should catch this error case (loading a GraphDef file that was created by a newer version of the framework) earlier. I filed #6120 to track that work.

**Comment by tfboyd** [comment]
Automatically closing due to lack of recent activity. We hope that you were able to resolve it on your own. However, since this is a support issue rather than a bug or feature request, you will probably get more information by posting it on StackOverflow.
